File: src/main/java/tutorial/model/Person0.java
package tutorial.model;

import jakarta.persistence.Column;
import jakarta.persistence.Convert;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import org.eclipse.persistence.annotations.IdValidation;
import org.eclipse.persistence.annotations.PrimaryKey;
import java.util.UUID;

@Entity
@Table(name = "person0", schema = "OLINGO")
public class Person0 {
    public Person0() {// Needed to be used as IdClass for java.lang.reflect.Constructor.newInstance
    }

    public Person0 (UUID _Id){
        userName = _Id;
    }

    @Id
    @Column(name = "userName")
    @Convert(converter = UUIDAttributeConverter.class)
    private UUID userName;

    @Column(name = "firstName", nullable = false, length = 20)
    String firstName;

    @Column(name = "lastName", nullable = false, length = 20)
    String lastName;

    public UUID getUserName() {
        return userName;
    }

    public void setUserName(UUID userName) {
        this.userName = userName;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }
}

File: src/main/java/tutorial/model/Person0.scala
package tutorial.model

import jakarta.persistence.{Column, Convert, Entity, Id, Table}
import java.util.UUID

import tutorial.model.Person0._Column

import scala.annotation.meta.{beanGetter, field}
import scala.beans.BeanProperty

@Entity
@Table(name = "person0", schema = "OLINGO")
class Person0 {
  @(Id@field)
  @Column(name = "userName")
  @Convert(converter = classOf[UUIDAttributeConverter])
  @BeanProperty
  var userName: UUID = _

  @(_Column@field)(name = "firstName", nullable = false, length = 20)
  @BeanProperty
  var firstName: String = _

  @(_Column@field)(name = "lastName", nullable = false, length = 20)
  @BeanProperty
  var lastName: String = _

  def this(_Id: UUID) = {
    this()
    userName = _Id
  }

}

object Person0 {
  type _Column = Column@beanGetter
  final val tableName = "person0"
  final val schemaName = "OLINGO"
}